### DISCUSSION TITLE
Next13 Pocketbase authstore

### DISCUSSION THREAD

**Discussion by jaywei1234** [pocketbase/pocketbase#1746] [Q&A]
I have a login system like this
![image](https://user-images.githubusercontent.com/56056463/216260681-e7b41582-7b0e-4b85-8c2f-91be07c501fd.png)

It has response after loggin
![image](https://user-images.githubusercontent.com/56056463/216260864-e7c5ae6c-7abc-4912-bfdf-a9a662b6803b.png)

But after the redirecting or refreshing, it seems the auth data didnt stored.

I've searched some similar discussions, but I cant find about the Next13.

What should I do in Next13 for the authstore. Thank you!

**Comment by ganigeorgiev** [comment]
Please check https://github.com/pocketbase/js-sdk/issues/69.

I still haven't got the time to check in details Next.js 13 (_as far as I know they still don't allow setting response cookies using the new app directory flow_).

When working with meta frameworks like SvelteKit, Next.js, Nuxt, etc. in general there are 3 ways to integrate with the SDK:

1. client-side only (your code runs only in the browser, eg. SPA)
2. server-side only (eg. SSR)
3. mixed client-side and server-side


The first option would be the simplest and it will work out of the box since the AuthStore by default uses the native LocalStorage to persist the auth data between redirects/reloads.

I suspect that you are using a SSR flow and this will require some other mechanism to persist the auth data, eg. cookies. For an example Nextjs.12 flow and some general SSR integration guidance you can check https://github.com/pocketbase/js-sdk#ssr-integration.

For 3), the situation is similar to the SSR flow, but you'll have to export the cookie with `httpOnly: false` and when in a browser env to load and set  the cookie using `document.cookie` instead of the server request/response objects (_some frameworks like Nuxt and SvelteKit has integrated abstraction to read and set cookies that could unify and simplify the process_).

**Reply by jaywei1234** [reply]
Thanks a lot, I will try it!

**Reply by jaywei1234** [reply]
I still can't find a solution for Next13 login. I jsut can't export the cookie by the using exportToCookie function.

**Reply by ganigeorgiev** [reply]
@jaywei1234 Without a context or at least code sample what you are doing, I'm not sure how to help.

Please provide a little more information about the error you've encountered or ideally a minimal reproducible repo, and I'll have a look at it, but as mentioned in https://github.com/pocketbase/js-sdk/issues/69, I still haven't played with Next.js 13 so I'm not sure what is the recommend approach there.

**Reply by jaywei1234** [reply]
![image](https://user-images.githubusercontent.com/56056463/220028752-b0fc11a2-5692-4854-801d-576cab14fe06.png)
Here is the login. I just add the exportToCookie function, but I can't find anything in the cookies after I login.

**Reply by ganigeorgiev** [reply]
@jaywei1234 Please check the examples in https://github.com/pocketbase/js-sdk#ssr-integration.

`exportToCookie` generates a cookie value/string. It doesn't write the cookie to the response - this is up to the developer and will differ depending on whether your code is executed on the node server-side or in the browser.

**Reply by jaywei1234** [reply]
Got it, thank you!!
